src/type.zig

```diff
@@ -6693,7 +6722,18 @@ pub const CType = enum {
             .windows, .uefi => switch (self) {
                 .short, .ushort => return 16,
                 .int, .uint, .long, .ulong => return 32,
-                .longlong, .ulonglong, .longdouble => return 64,
+                .longlong, .ulonglong => return 64,
+                .longdouble => switch (target.cpu.arch) {
+                    .i386 => switch (target.abi) {
```

Comment:
Does this need a corresponding update to `longDoubleIs`?

Reply:
Looks like it does, thanks for pointing that out!

Reply:
No problem - This inspired me to write a quick test actually: https://zigbin.io/c3ccdd

Feel free to use or reimplement, if you think the test is worthwhile. Otherwise, I'll probably roll this into one of my own PRs to help in the future.

Reply:
Test looks good; might be better for a subsequent PR since it looks like a lot of other ABIs might be using incorrect long double info?

Reply:
Sounds good to me. I'll do a fix-up pass soon.

Thanks for fixing up i386!